### ISSUE TITLE
missing-resources calculation wrong - can measure negative value

### ISSUE THREAD

**Issue by sslavic** [allegro/hermes#635]
If my understanding is correct, missing-resources sample measurement should be either 0 or positive integer. What we're seeing is that negative value gets measured.

Here's how to reproduce the issue - have one topic, one subscription, selective workload algorithm, and workload set to 1 consumer per subscription, with two hermes-consumers instances in the cluster.

**Comment by wojtkiewicz** [comment]
Missing resources are calculated as a sum of map keys and some map implementations can throw -1 when asked for their size(), I will look into this

**Comment by sslavic** [comment]
After recent redeployments of our hermes clusters, we're not seeing issue with missing-resources negative value calculation. Instead, we're seeing now BalancingJob is failing with
```
java.util.NoSuchElementException: No value present at java.util.Optional.get(Optional.java:135) at pl.allegro.tech.hermes.consumers.supervisor.workload.selective.SelectiveWorkBalancer.maxLoadedConsumerNode(SelectiveWorkBalancer.java:84)
...
```

Thinking there is same common cause to both previous wrong calculation and this error being thrown now.

We have blue-green deployment pipeline of hermes-consumers, so when we need to make a new deployment e.g. of a new release, with existing hermes-consumers (blue) cluster running, we deploy along with it a new (green) cluster using new release. Both blue and green cluster use same ZooKeeper/Kafka cluster, and are up for a brief period of time, until deployment pipeline determines that green deployment has successfully started. If that was the case, old, blue cluster gets stopped, and green one remains running. Something in hermes-consumers is not always dealing well with this change, so at lest for some time keeps references to non-existing nodes, consumer node or subscription assignments they used to have but are no longer part of the cluster.

**Comment by adamdubiel** [comment]
I think this should be fixed by #664 : closing. Let me know if something is still not right.
